File: src/main/java/net/reflact/client/hud/overlay/HotbarOverlay.java
package net.reflact.client.hud.overlay;

import net.minecraft.client.MinecraftClient;
import net.minecraft.client.gui.DrawContext;
import net.minecraft.client.network.ClientPlayerEntity;
import net.minecraft.client.render.RenderLayer;
import net.minecraft.item.ItemStack;
import net.minecraft.util.Identifier;
import net.reflact.client.ReflactClient;

public class HotbarOverlay implements ReflactOverlay {
    private static final Identifier TEXTURE = Identifier.of("reflact", "textures/gui/sprites/hud/hotbar.png");

    @Override
    public void render(DrawContext context, float tickDelta) {
        MinecraftClient client = MinecraftClient.getInstance();
        ClientPlayerEntity player = client.player;
        if (player == null) return;

        int x = getX();
        int y = getY();
        int width = getWidth();
        int height = getHeight();

        // Draw Background (Polished: Translucent Black with Border)
        // Main background
        context.fill(x, y, x + width, y + height, 0xAA000000);
        // Top/Bottom borders
        context.fill(x, y, x + width, y + 1, 0xFF555555);
        context.fill(x, y + height - 1, x + width, y + height, 0xFF555555);
        // Left/Right borders
        context.fill(x, y, x + 1, y + height, 0xFF555555);
        context.fill(x + width - 1, y, x + width, y + height, 0xFF555555);

        // Draw Slots
        for (int i = 0; i < 9; i++) {
            ItemStack stack = player.getInventory().getStack(i);
            int itemX = x + 3 + i * 20; 
            int itemY = y + 3; 
            
            // Slot background (slightly lighter)
            context.fill(itemX - 1, itemY - 1, itemX + 17, itemY + 17, 0x44FFFFFF);
            
            context.drawItem(stack, itemX, itemY);
            context.drawStackOverlay(client.textRenderer, stack, itemX, itemY);
            
            // Draw selection
            if (player.getInventory().getSelectedSlot() == i) {
                 // Selection Border (White, thicker)
                 context.fill(itemX - 2, itemY - 2, itemX + 18, itemY - 1, 0xFFFFFFFF); // Top
                 context.fill(itemX - 2, itemY + 17, itemX + 18, itemY + 18, 0xFFFFFFFF); // Bottom
                 context.fill(itemX - 2, itemY - 1, itemX - 1, itemY + 17, 0xFFFFFFFF); // Left
                 context.fill(itemX + 17, itemY - 1, itemX + 18, itemY + 17, 0xFFFFFFFF); // Right
            }
        }
    }

    @Override
    public int getX() { 
        int cfg = ReflactClient.CONFIG.hotbarX();
        if (cfg == 0) {
            MinecraftClient client = MinecraftClient.getInstance();
            return client.getWindow().getScaledWidth() / 2 - 91; 
        }
        return cfg;
    }
    
    @Override
    public int getY() { 
        int cfg = ReflactClient.CONFIG.hotbarY();
        if (cfg == 0) {
            MinecraftClient client = MinecraftClient.getInstance();
            return client.getWindow().getScaledHeight() - 22; 
        }
        return cfg;
    }

    @Override public void setX(int x) { ReflactClient.CONFIG.hotbarX(x); }
    @Override public void setY(int y) { ReflactClient.CONFIG.hotbarY(y); }
    @Override public int getWidth() { return 182; }
    @Override public int getHeight() { return 22; }
    @Override public boolean isEnabled() { return ReflactClient.CONFIG.showHotbar(); }
    @Override public String getName() { return "Hotbar"; }
}

File: src/main/kotlin/net/reflact/client/hud/overlay/HotbarOverlay.kt
package net.reflact.client.hud.overlay

import net.minecraft.client.MinecraftClient
import net.minecraft.client.gui.DrawContext
import net.minecraft.util.Identifier
import net.reflact.client.ReflactClient

class HotbarOverlay : ReflactOverlay {
    override fun render(context: DrawContext, tickDelta: Float) {
        val client = MinecraftClient.getInstance()
        val player = client.player ?: return

        val x = x
        val y = y
        val width = width
        val height = height

        // Draw Background (Polished: Translucent Black with Border)
        // Main background
        context.fill(x, y, x + width, y + height, 0xAA000000.toInt())
        // Top/Bottom borders
        context.fill(x, y, x + width, y + 1, 0xFF555555.toInt())
        context.fill(x, y + height - 1, x + width, y + height, 0xFF555555.toInt())
        // Left/Right borders
        context.fill(x, y, x + 1, y + height, 0xFF555555.toInt())
        context.fill(x + width - 1, y, x + width, y + height, 0xFF555555.toInt())

        // Draw Slots
        for (i in 0 until 9) {
            val stack = player.inventory.getStack(i)
            val itemX = x + 3 + i * 20
            val itemY = y + 3

            // Slot background (slightly lighter)
            context.fill(itemX - 1, itemY - 1, itemX + 17, itemY + 17, 0x44FFFFFF)

            context.drawItem(stack, itemX, itemY)
            context.drawStackOverlay(client.textRenderer, stack, itemX, itemY)

            // Draw selection
            if (player.inventory.selectedSlot == i) {
                // Selection Border (White, thicker)
                context.fill(itemX - 2, itemY - 2, itemX + 18, itemY - 1, 0xFFFFFFFF.toInt()) // Top
                context.fill(itemX - 2, itemY + 17, itemX + 18, itemY + 18, 0xFFFFFFFF.toInt()) // Bottom
                context.fill(itemX - 2, itemY - 1, itemX - 1, itemY + 17, 0xFFFFFFFF.toInt()) // Left
                context.fill(itemX + 17, itemY - 1, itemX + 18, itemY + 17, 0xFFFFFFFF.toInt()) // Right
            }
        }
    }

    override var x: Int
        get() {
            val cfg = ReflactClient.CONFIG.hotbarX()
            if (cfg == 0) {
                val client = MinecraftClient.getInstance()
                return client.window.scaledWidth / 2 - 91
            }
            return cfg
        }
        set(value) { ReflactClient.CONFIG.hotbarX(value) }

    override var y: Int
        get() {
            val cfg = ReflactClient.CONFIG.hotbarY()
            if (cfg == 0) {
                val client = MinecraftClient.getInstance()
                return client.window.scaledHeight - 22
            }
            return cfg
        }
        set(value) { ReflactClient.CONFIG.hotbarY(value) }

    override var width: Int get() = 182; set(value) {}
    override var height: Int get() = 22; set(value) {}
    override val isEnabled: Boolean get() = ReflactClient.CONFIG.showHotbar()
    override val name: String get() = "Hotbar"

    companion object {
        private val TEXTURE = Identifier.of("reflact", "textures/gui/sprites/hud/hotbar.png")
    }
}
